crates/rbuilder/src/backtest/fetch/datasource.rs

```diff
@@ -0,0 +1,32 @@
+use crate::backtest::OrdersWithTimestamp;
+use async_trait::async_trait;
+
+/// Datasource trait
+///
+/// This trait is used to fetch data from a datasource
+#[async_trait]
+pub trait Datasource: std::fmt::Debug {
+    async fn get_data(&self, block: BlockRef) -> eyre::Result<Vec<OrdersWithTimestamp>>;
+    fn clone_box(&self) -> Box<dyn Datasource>;
```

Comment:
Why this instead of making Datasource Clonable (the same way it has std::fmt::Debug)?

Reply:
I tried it but it does not work. Do not have enough context in Rust to know. This was generated by Claude but also a common pattern I found for this use case of storing vec<box<dyn Trait>>>

Reply:
dyn can't be used with Clonable since it's Sized :(.